velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -515,37 +582,13 @@ TEST_P(PrestoSerializerTest, longDecimal) {
 // Test that hierarchically encoded columns (rows) have their encodings
 // preserved.
 TEST_P(PrestoSerializerTest, encodings) {
-  auto baseNoNulls = makeFlatVector<int64_t>({1, 2, 3, 4});
-  auto baseWithNulls = makeNullableFlatVector<int32_t>({1, std::nullopt, 2, 3});
-  auto baseArray =
-      makeArrayVector<int32_t>({{1, 2, 3}, {}, {4, 5}, {6, 7, 8, 9, 10}});
-  auto indices = makeIndices(8, [](auto row) { return row / 2; });
-
-  auto data = makeRowVector({
-      BaseVector::wrapInDictionary(nullptr, indices, 8, baseNoNulls),
-      BaseVector::wrapInDictionary(nullptr, indices, 8, baseWithNulls),
-      BaseVector::wrapInDictionary(nullptr, indices, 8, baseArray),
-      BaseVector::createConstant(INTEGER(), 123, 8, pool_.get()),
-      BaseVector::createNullConstant(VARCHAR(), 8, pool_.get()),
-      BaseVector::wrapInConstant(8, 1, baseArray),
-      BaseVector::wrapInConstant(8, 2, baseArray),
-      makeRowVector({
-          BaseVector::wrapInDictionary(nullptr, indices, 8, baseNoNulls),
-          BaseVector::wrapInDictionary(nullptr, indices, 8, baseWithNulls),
-          BaseVector::wrapInDictionary(nullptr, indices, 8, baseArray),
-          BaseVector::createConstant(INTEGER(), 123, 8, pool_.get()),
-          BaseVector::createNullConstant(VARCHAR(), 8, pool_.get()),
-          BaseVector::wrapInConstant(8, 1, baseArray),
-          BaseVector::wrapInConstant(8, 2, baseArray),
-          makeRowVector({
-              BaseVector::wrapInDictionary(nullptr, indices, 8, baseWithNulls),
-              BaseVector::createConstant(INTEGER(), 123, 8, pool_.get()),
-              BaseVector::wrapInConstant(8, 2, baseArray),
-          }),
-      }),
-  });
+  testEncodedRoundTrip(encodingsTestVector());
+}
 
-  testEncodedRoundTrip(data);
+// Test that hierarchically encoded columns (rows) have their encodings
+// preserved by the PrestoBatchVectorSerializer.
+TEST_P(PrestoSerializerTest, encodingsBatchVectorSerializer) {
+  testBatchVectorSerializerRoundTrip(encodingsTestVector());
```

Comment:
Is it possible make the serializer type as a test parameter in followup? Thanks!